### ISSUE TITLE
[BUG] toHaveScreenshot captures blank screenshots, even though waitFor is used

### ISSUE THREAD

**Issue by ivan-krupskyi** [microsoft/playwright#21657]
### System info
- Playwright Version: v1.31.2
- Operating System: Linux
- Browser: All

- Description:
    - I am trying to capture "golden" screenshots for my visual regression tests. I have implemented a `locator.waitFor({state:'visible'})` to wait for page to be stable after navigation (Also tried to wait with `await expect(locator).toBeVisible()` method). And only after that to capture a screenshot. waitFor/expect().toBeVisible() methods are passing - meaning page is loaded and anchor element is visible, but toHaveScreenshots method is capturing blank screenshots anyways.

**please see attached video of test execution**
[video.webm](https://user-images.githubusercontent.com/102296081/224996447-a8e2bfa2-9afb-43fd-9807-41de232c678c.webm)

You'll see blinking screen **(from 9sec-10sec in the video)** - that is the moment of screenshot capturing. Why does it capture screenshot while page is still blank ?
Is locator.waitFor({state:'visible'}) method not working correctly?

**test itself** (very simple test, navigate to the page, wait for "anchor" element to be visible and capture a screenshot):
`test('Visual comparison of sign up page', async ({page, isMobile}) => {`
        `   const signUpPage = new SignUpPage(page, isMobile)`
       `   await signUpPage.gotoSignUp();` --> should navigate to sign up page and wait for "create" button to be visible
      `   await visualComparisonHelper.visualComparison({clipFooter: false});` --> triggers toHaveScreenshot() method
    `});`

await signUpPage.gotoSignUp() method:
`async gotoSignUp() {`
        `   await this.page.goto('/account/register')`
        `   await expect(this.createButton).toBeVisible()` -- waiting for this element to appear which would mean that page is loaded
    `}`

I have tried to wait for element with `await this.createButton.waitFor({state: 'visible'})` - that didn't work either

- Actual result:
   - Captures blank screenshot

- Expected result:
   - Captures screenshot of the page

**Comment by ivan-krupskyi** [comment]
For now to solve the problem I have added a static wait.
This solves the problem. 
But it supposed to work without that, as I am waiting for element to appear on the page already in await signInPage.gotoSignIn() method.

`test('Visual comparison of sign in page', async ({page, isMobile}) => {`
       `const signInPage = new SignInPage(page, isMobile)`
        `   await signInPage.gotoSignIn()`
       `   await page.waitForTimeout(5000)` **--> STATIC WAIT HERE**
      `   await visualComparisonHelper.visualComparison({clipFooter: false});`
  `  });`

**Comment by ivan-krupskyi** [comment]
I can't unfortunately share the page, because it's internal company environment. But let me know if I can help in any other way

**Comment by dgozman** [comment]
Unfortunately, we have to reproduce this issue locally to be able to debug it. If you can provide a full repro, please file a new issue with it, and link to this one.

**Comment by jpratt2** [comment]
reopening here: https://github.com/microsoft/playwright/issues/32040
